source/scripts/user.js

```diff
@@ -0,0 +1,121 @@
+ * login to a user account that help manipulate the account system. A User class might not be needed
+ * to create a general object for a user to be stored in local storage.
+ * -------------------------------------
+ * Some stretch goals:
+ * - delete an account and delete all the posts on that account (?)
+ * - retype password to confirm when creating an account
+ * - "Forgot password" option in login page
+ * - when someone enters the page containg the domain of our app, redirect to login page if not logged in
+ */
+
+// constant urls
+const APP_MAIN_URL = "some_url_for_the_main_app_page";
+const CREATE_ACCOUNT_URL = "some_url_for_user_to_create_account";
+const LOGIN_URL = "some_url_for_user_to_login_in";
+
+/**
+ * Class for creating User type Object, could instead just declare a general Object type without this
+ */
+class User extends HTMLElement() {
+    constructor(){
+        super();
+        // to add something that needed to be displayed on html page
+    }
+    set data(data){
+        // data that need to be visualized on html page
+    }
+}
+
+// customElements.define('user-account', User); // custom element from User Object, might not be needed
+
+/**
+ * Get the users from local storage as an array of user objects
+ */
+function get_users_from_storage() {
+    if (!localStorage.getItem('users')) {
+      let empty_array = [];
+      return empty_array;
+    }
+    return JSON.parse(localStorage.getItem('users'));
+}
+
+/**
+ * Takes in an array of users, converts it to a string, and then
+ * saves that string to 'users' in localStorage
+ */
+function save_users_to_storage(users) {
+    let users_string = JSON.stringify(users);
+    localStorage.setItem('users', users_string);
+}
+
+/**
+ * Event handler to <form> of creating a user account
+ */
+function create_account_form_handler() {
+    let user_object = {};
+    let user_info = document.querySelector('.create_account'); // need to be created in createaccount.html
+    let form_data = new FormData(user_info); // form data from user input
+    let username = form_data.username.value;
+    let password = form_data.password.value;
+    
+    let submit_button = document.querySelectorAll('button'); // type 'button' in createaccount.html
+    submit_button.addEventListener('click', create_account);
+
+    function create_account() {
+        // first check if an account associated with the username exists
+        if (localStorage.getItem(username) != null) {
+            alert('User already exists!')
+            console.log('Fail to create a new account');
+
+            // redirect to login page
+            location.replace(LOGIN_URL);
+            return;
+        }
+        // add new user to local storage       
+        user_object['username'] = form_data.username.value;
+        user_object['password'] = form_data.password.value;
+        let users_array = get_users_from_storage();
+        users_array.push(user_object);
+        save_users_to_storage(users_array);

```

Comment:
There is a discrepancy between how you are checking accounts and how you are storing accounts.  On one hand you are checking for accounts by searching there username into local storage, which is the way we discussed in the ADR but you are storing them by pushing an array of users_objects.  I think storing as an array like that is better so you should keep that, but you will need to change the way you check if an account exists.